artiq/gateware/suservo/iir.py

```diff
@@ -7,18 +7,20 @@
 
 # all these are number of bits!
 IIRWidths = namedtuple("IIRWidths", [
-    "state",    # the signed x and y states of the IIR filter
-                # DSP A input, x state is one bit smaller
-                # due to AD pre-adder, y has full width (25)
-    "coeff",    # signed IIR filter coefficients a1, b0, b1 (18)
-    "accu",     # IIR accumulator width (48)
-    "adc",      # signed ADC data (16)
-    "word",     # "word" size to break up DDS profile data (16)
-    "asf",      # unsigned amplitude scale factor for DDS (14)
-    "shift",    # fixed point scaling coefficient for a1, b0, b1 (log2!) (11)
-    "channel",  # channels (log2!) (3)
-    "profile",  # profiles per channel (log2!) (5)
-    "dly",      # the activation delay
+    "state",        # the signed x and y states of the IIR filter
+                    # DSP A input, x state is one bit smaller
+                    # due to AD pre-adder, y has full width (25)
+    "coeff",        # signed IIR filter coefficients a1, b0, b1 (18)
+    "accu",         # IIR accumulator width (48)
+    "adc",          # signed ADC data (16)
+    "word",         # "word" size to break up DDS profile data (16)
+    "asf",          # unsigned amplitude scale factor for DDS (14)
+    "shift",        # fixed point scaling coefficient for a1, b0, b1 (log2!) (11)
+    "adc_channels", # number of input ADC channels (8)
+    "dds_channels", # number of output DDS channels (8)
+                    # the number of IIR channels is matched to this
```

Comment:
Does `IIRWidths` now mix both bit widths as well as the number of channels? Not only are they not relaly related; mixing numbers and log2/bit-width seems error prone. This is why we used `ADCParams` (which already has a `channels` field) and `DDSParams` in #1782.